src/ai/ai.cpp

```diff
@@ -1800,12 +1800,19 @@ void update_cb_fabrication(sys::state& state) {
 }
 
 bool will_join_war(sys::state& state, dcon::nation_id n, dcon::war_id w, bool as_attacker) {
+	// Prevent constraint violations
+	if(!military::standard_war_joining_is_possible(state, w, n, as_attacker))
```

Comment:
this check should be made in the diplomatic action acceptance, rather than here, otherwise players may be able to join invalid wars